tests/src/Kernel/ViewsLinkFieldsTest.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\collabora_online\Kernel;
+
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\media\Entity\Media;
+use Drupal\Tests\collabora_online\Traits\MediaCreationTrait;
+use Drupal\Tests\media\Traits\MediaTypeCreationTrait;
+use Drupal\Tests\user\Traits\UserCreationTrait;
+use Drupal\views\Views;
+
+/**
+ * Tests link fields to preview and edit medias in views.
+ */
+class ViewsLinkFieldsTest extends KernelTestBase {
+
+    use UserCreationTrait;
+    use MediaCreationTrait;
+    use MediaTypeCreationTrait;
+
+    /**
+     * Media owned by current user.
+     *
+     * @var \Drupal\media\MediaInterface;
+     */
+    protected $ownMedia = NULL;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected static $modules = [
+        'collabora_online',
+        'collabora_online_test',
+        'field',
+        'file',
+        'image',
+        'media',
+        'system',
+        'user',
+        'views',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp(): void {
+        parent::setUp();
+
+        $this->installEntitySchema('file');
+        $this->installEntitySchema('media');
+        $this->installEntitySchema('user');
+        $this->installConfig(['user', 'views', 'collabora_online_test']);
+        $this->installSchema('file', ['file_usage']);
+        // Install user module avoid user 1 permissions bypass.
+        \Drupal::moduleHandler()->loadInclude('user', 'install');
+        user_install();
+
+        // Create two medias to check access with different scopes, 'any' and 'own'.
+        $this->createMediaEntity('document');
+        $this->ownMedia = $this->createMediaEntity('document');;
+    }
+
+    /**
+     * Tests link fields.
+     */
+    public function testLinks(): void {
+        // User without permissions can't see links.
+        $this->doTestLinks([
+            'preview' => [FALSE, FALSE],
+            'edit' => [FALSE, FALSE],
+        ],
+        $this->createUser([]));
+        // User with 'Preview' permission can see preview link.
+        $this->doTestLinks([
+            'preview' => [TRUE, TRUE],
+            'edit' => [FALSE, FALSE],
+        ],
+        $this->createUser([
+        'preview document in collabora'
+        ]));
```

Comment:
The line breaks are off, we don't clearly see where one call ends and the other begins.
```suggestion
        $this->doTestLinks(
            [
                'preview' => [FALSE, FALSE],
                'edit' => [FALSE, FALSE],
            ],
            $this->createUser([]),
        );
        // User with 'Preview' permission can see preview link.
        $this->doTestLinks(
            [
                'preview' => [TRUE, TRUE],
                'edit' => [FALSE, FALSE],
            ],
            $this->createUser(['preview document in collabora']),
        );
```
Also review the other calls below.

Reply:
Fixed with other format issues.